[BB-2500] OCIM developer frontend documentation for New Console (#586)
open-craft/opencraft: frontend/README.md

A single-page app for the [OpenCraft Instance Manager (OCIM)](https://github.com/open-craft/opencraft).

> ## Provisioning

Reviewer: ```suggestion
## Running locally
```

Reviewer: Also, add a small introduction here that this should be run outside the Vagrant devstack to ease the use since it's just an independent React SPA.

- Install the API client:

```bash
./scripts/build-api-client.sh
```

- Install requirements:

```bash
npm install
```

- Run frontend:

```bash
npm start
```

- Generating New Components:

```bash
npm run generate # uses plop, more information here(https://plopjs.com/)
```

## API Client Instructions

- Updating the API Client:

```bash
cd ./frontend # go to frontend directory
npm run update-api-client
```

- Building the API Client:

```bash
cd ./frontend # go to frontend directory
npm run build-api-client
```

## Testing

> - Running the tests:

Reviewer: ```suggestion
- Running tests and checks:
```

Author: Thanks, I added it

```bash
npm run lint # for running the linting

> npm run lint-fix # for fixing the linting

Reviewer: This just runs eslint with the `--fix` flag so it tries to correct as many issues as it can, but it'll still fail in case it doesn't manage to fix something.

Author: Added the details in the comments to make it more clear

npm run test # for running the tests

> make test.js && make test.frontend && npm run lint # for running the test using the makefile

Reviewer: This is not accurate. I don't think anyone should run tests like this.
Either list the available `make` commands for testing or remove this entirely.

Author: I have removed them all together as most of them usually runs whenever we run `make test`

```

## Deployment Process

> The Deployment been done on the master automatically using CircleCI(for details, check `frontend-deploy` job in [circle.yml](../circle.yml)). The bundle been hosted on S3 and been directly served from the Cloudfront.

Reviewer: ```suggestion
The Deployment is done on automatically using CircleCI when any commit is added in the `master` branch (for details, check `frontend-deploy` job in [circle.yml](../circle.yml)). The bundle is hosted on S3 and is directly served from Cloudfront.
```

Author: Thanks for the suggestion :), I have added it

## Running with Vagrant

> While it is possible to run this frontend server within Vagrant during

Reviewer: Move this section up as I've mentioned in my other comment. This doesn't need to be inside a section, and can just be a paragraph in the Running locally section.

Author: Yeah, I have removed it altogether and added a text inside the running locally section.

development, for performance reasons it's better to run it separately
outside of Vagrant instead.

## Reuseable UI Components

> All reusable UI components are described in the `/demo` route, accessible in development environment through [localhost](http://localhost:3000/demo).

Reviewer: Not all components are listed there.
```suggestion
A partial list of reusable UI components is shown in the `/demo` route, accessible in the development environment through [http://localhost:3000/demo](http://localhost:3000/demo).
```

Author: Thanks for the suggestion, I have added it.

# Provisioning

1. Install the API client:

```bash
./scripts/build-api-client.sh
```

2. Install requirements:

```bash
npm install
```

3. Run frontend:

```bash
npm start

> ```

Author: Thanks for the suggestions, I have added the separate sections now :+1:

4. Updating the API Client:

```bash
cd ./frontend # go to frontend directory
npm run update-api-client
```

5. Building the API Client:

```bash
cd ./frontend # go to frontend directory
npm run build-api-client
```

6. Generating New Components:

This 

```bash
npm run generate # uses plop, more information here(https://plopjs.com/)
```

7. Running the tests:

```bash
npm run lint # for running the linting

npm run lint-fix # for fixing the linting

npm run test # for running the tests

make test.js && make test.frontend && npm run lint # for running the test using the makefile
```

> 8. Deployment:

Author: Done, added the link to `circle.yml`

The Deployment been done on the master automatically using CircleCI. The bundle been hosted on S3 and been directly served from the Cloudfront.

# Running with Vagrant

While it is possible to run this frontend server within Vagrant during
development, for performance reasons it's better to run it separately
outside of Vagrant instead.

# Reuseable UI Components

> All reusable UI components description been described in the `/demo`

Author: Thanks for the suggestions, I have added them :)

> route in dev environment.

Author: Thanks for this, I have changed them :)
